docs/custom-code/prisma-custom-attributes.md

```diff
@@ -0,0 +1,314 @@
+
+Amplication will generate the following Prisma schema:
+
+```ts
+datasource mysql {
+  provider = "mysql"
+  url      = env("DB_URL")
+}
+
+generator client {
+  provider = "prisma-client-js"
+}
+
+model User {
+  createdAt DateTime @default(now())
+  firstName String?
+  id        String   @id @default(cuid())
+  lastName  String?
+  password  String
+  roles     Json
+  updatedAt DateTime @updatedAt
+  username  String   @unique
+}
+```
+
+:::important
+The User entity is _only_ created for your Amplication service if you enable authentication.
+:::
+
+The above code snippet is an example of the `schema.prisma` file that Amplication automatically generates for your application.
+Any modifications that you make to your project's _Entities_ and their _Fields_ get reflected as changes in your Prisma schema.
+
+Now let's say you add some new fields to your service's _User_ entity.
+For example, a new `projects` field and a `tasks` field.
+
+After you commit your changes and Amplication builds your project, the newly generated `schema.prisma` file will look like the following:
+
+```ts
+model User {
+  createdAt DateTime  @default(now())
+  firstName String?
+  id        String    @id @default(cuid())
+  lastName  String?
+  password  String
+  projects  Project[]
+  roles     Json
+  tasks     Task[]
+  updatedAt DateTime  @updatedAt
+  username  String    @unique
+}
+```
+
+:::important
+The [Building New Versions of Your Service](/building-new-versions/) shows you the necessary steps for adding new entities and fields to your service and [committing those changes to your git provider](/sync-with-github/).
+:::
+
+All of the changes that you make through Amplication's UI always translate back to changes in your code, in this case (database changes), to the generated `schema.prisma` file.
+
+**But, Prisma is powerful. It allows you to further refine and customize your schema beyond what Amplication allows you to do purely through the UI.**
+
+This is why Amplication gives you access to the Custom Attributes text field to fully access that feature and further customize your entities
+
+## What are Prisma Custom Attributes?
+
+Prisma's Custom Attributes allow you to modify the behavior of an entire Entity or a specific Field.
+
+Let's go through a few examples of when Custom Attributes prove to be useful:
+
+1. **Make sure usernames are unique**: When a user signs up for your application, you want to ensure that each user has a unique username.
+Using Prisma's custom attributes you can enforce this constraint at the field level.
+
+```ts
+model User {
+  username String @unique
+}
+```
+
+:::info
+This is a simplified schema to showcase Prisma's `@unique` [field-level custom attribute](#field-level-custom-attributes).
+Amplication supports the `@unique` out of the box directly from the UI.
+But, if you use the unique field toggle, do not include `@unique` in the _Custom Attributes_ textarea.
+You will see errors in your schema.
```

Comment:
I guess you mean "otherwise, you will see errors" as continuation of the previous sentence?

Reply:
Thanks, @mulygottlieb, I implemented this feedback. All other feedback should be implemented, but please let me know if I can update anything.